server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/MailDTOTest.java

```diff
@@ -0,0 +1,96 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Instant;
+import java.util.Date;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.mail.MimeMessagePartsId;
+import org.apache.mailet.PerRecipientHeaders;
+import org.apache.mailet.base.MailAddressFixture;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+class MailDTOTest {
+    static final HashBlobId.Factory BLOB_ID_FACTORY = new HashBlobId.Factory();
+    static final Date LAST_UPDATED = Date.from(Instant.parse("2016-09-08T14:25:52.000Z"));
+
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setUp() {
+        objectMapper = new ObjectMapper()
+            .registerModule(new Jdk8Module())
+            .registerModule(new JavaTimeModule())
+            .registerModule(new GuavaModule());
+    }
+
+    @Test
+    void mailDtoShouldBeSerializedToTheRightFormat() throws Exception {
+        assertThatJson(objectMapper.writeValueAsString(mailDTO1()))
+            .isEqualTo(getSystemResourceAsString("json/mail1.json"));
+    }
+
+    @Test
+    void mailDtoShouldBeDeserializeFromTheRightFormat() throws Exception {
+        assertThat(objectMapper.readValue(getSystemResourceAsString("json/mail1.json"), MailDTO.class))
+            .isEqualTo(mailDTO1());
+    }
+
+    private MailDTO mailDTO1() throws MessagingException {
+        return MailDTO.fromMail(
+            FakeMail.builder()
+                .recipients(MailAddressFixture.RECIPIENT1, MailAddressFixture.RECIPIENT2)
+                .sender(MailAddressFixture.SENDER)
+                .attribute("att1", "value")
+                .errorMessage("an error")
+                .lastUpdated(LAST_UPDATED)
+                .remoteHost("toto.com")
+                .remoteAddr("159.221.12.145")
+                .addHeaderForRecipient(PerRecipientHeaders.Header.builder()
+                    .name("X-custom-header")
+                    .value("uedcgukrcg")
+                    .build(), MailAddressFixture.RECIPIENT1)
+                .addHeaderForRecipient(PerRecipientHeaders.Header.builder()
+                    .name("X-custom-header-2")
+                    .value("uedcgukrcg")
+                    .build(), MailAddressFixture.RECIPIENT2)
+                .state("state")
+                .name("mail-name-558")
+                .build(),
+            MimeMessagePartsId.builder()
+                .headerBlobId(BLOB_ID_FACTORY.from("210e7136-ede3-44eb-9495-3ed816d6e23b"))
+                .bodyBlobId(BLOB_ID_FACTORY.from("ef46c026-7819-4048-b562-3a37469191ed"))
+                .build());
+    }
```

Comment:
maybe you can add a test with a minimalist mail (with all non mandatory fields empty)